### PR TITLE
add support for -shotdir parameter, fall back to -save parm or current dir

### LINKED ISSUE #587: add support for "-shotdir" command line parameter
I've found myself using Woof more and more these days, so I'm starting to notice all the small differences between this and other ports I was accustomed to...
The option to write screenshots to a custom directory is one of them: I think it would be useful for people that want to keep the program directory tidy.

Could this be implemented? 
As an alternative, writing screenshots to the custom savegame directory (when specified) would work too...

For reference
https://doomwiki.org/wiki/Source_port_parameters#-shotdir_.3Cdirectory.3E

Thank you!

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Fixes #587

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
Seriously considering to rip out support for PCX screenshots entirely...

**Comment by rfomin** [comment]
> Seriously considering to rip out support for PCX screenshots entirely...

Yes please.

File: Source/d_main.c [hunk 1]
```diff
@@ -859,18 +860,40 @@ void IdentifyVersion (void)
...
+  if (i > 0)
+  {
+    if (!stat(myargv[i + 1], &sbuf) && S_ISDIR(sbuf.st_mode))
+    {
+      if (screenshotdir)
+        free(screenshotdir);
+      screenshotdir = M_StringDuplicate(myargv[i + 1]);
+    }
+    else
+      puts("Error: -shotdir path does not exist, using default dir");
```
Comment (rfomin) [inline Source/d_main.c:894]: What if we call `M_MakeDirectory()` here? And for `-save` parameter too.

**Review by rfomin — APPROVED** [review]
Works on Windows.

**Pull request merged**